tests/test_single_topology_v3.py

```diff
@@ -383,3 +390,113 @@ def test_no_chiral_bond_restraints():
     assert len(state.chiral_bond.get_idxs()) == 0
     U = state.get_U_fn()
     _ = U(init_conf)
+
+
+def assert_interpolation_valid(f, a, b):
+    np.testing.assert_allclose(f(a, b, 0.0), a)
+    np.testing.assert_allclose(f(a, b, 1.0), b)
+
+    rng = np.random.default_rng(2022)
+    lambdas = rng.uniform(0, 1, (30,))
+    ys = vmap(functools.partial(f, a, b))(lambdas)
+    assert np.all(np.minimum(a, b) <= ys)
```

Comment:
is there a stronger condition we can check? i.e. not just that min(a,b) <= for all f(a,b,lambdas) <= max(a,b) but that it's monotonically increasing?

Reply:
This is now checked here: https://github.com/proteneer/timemachine/pull/820/files#diff-e679449b0c7d9911ad6910fd3cac904a8aaa1af2b02ad5d7873de170daa291bcR448-R450